src/test/java/com/training/services/tests/ProductsTest.java

```diff
@@ -62,8 +66,88 @@ public void testShorexProductTypeValidate() {
+
+        Map<String, Integer> day_Month_daysInMonthFromLink = getDay_Month_daysInMonth(map.get("sailingIdParamValue").split("AL")[1]);
+
+        for (int index = 0; index < response.getPayload().getProducts().size(); index++) {
+            for (int getDateOffering = 0; getDateOffering < response.getPayload().getProducts().get(index).getOffering().size(); getDateOffering++) {
+
+                Map<String, Integer> day_Month_daysInMonth = getDay_Month_daysInMonth(response.getPayload().getProducts().get(index).getOffering().get(getDateOffering).getOfferingDate());
+
+                /**
+                 * If query paramter link date and offeringDate are from same month
+                 */
+                if (day_Month_daysInMonthFromLink.get("month") == day_Month_daysInMonth.get("month")) {
+                    if (!((day_Month_daysInMonth.get("day") - day_Month_daysInMonthFromLink.get("day")) <= 10)) {
+                        softAssert.fail("Product ID = " + response.getPayload().getProducts()
+                                .get(index).getOffering().get(getDateOffering)
+                                .getProductID() + " offering date is greater than 10 days = "
+                                + response.getPayload().getProducts().get(index).getOffering().get(getDateOffering).getOfferingDate());
+                    }
+                } else {
+                    softAssert.assertThat(
+                            (day_Month_daysInMonthFromLink.get("daysInMonth") - day_Month_daysInMonthFromLink.get("day")) + day_Month_daysInMonth.get("day") <= 10)
+                            .as("Product ID = "
+                                    + response.getPayload().getProducts().get(index).getOffering().get(getDateOffering)
+                                    .getProductID() + " offeringDate is greater than 10 days = "
+                                    + response.getPayload().getProducts().get(index).getOffering().get(getDateOffering)
+                                    .getOfferingDate())
+                            .isTrue();
+                }
+            }
+        }
+    }
+
+    /**
+     * Returning segregate day, month and number of days in a month
+     *
+     * @param actualDateFromResponse
+     * @return
+     * @throws ParseException
+     */
+    public Map<String, Integer> getDay_Month_daysInMonth(String actualDateFromResponse) throws ParseException {
+        Map<String, Integer> dateMap = new HashMap<>();
+
+        /**
+         * actualDateInWholeText =  = Sun Aug 09 00:00:00 IST 2020
+         */
+        SimpleDateFormat originalFormat = new SimpleDateFormat("yyyyMMdd");
+        Date actualDateWholeText = originalFormat.parse(actualDateFromResponse);
+
+        /**
+         * Segregate date = 2020-08-09
+         */
+        SimpleDateFormat newFormat = new SimpleDateFormat("yyyy-MM-dd");
+        String segregateDate = newFormat.format(actualDateWholeText);
+        String dateSplit[] = segregateDate.split("-");
```

Comment:
Lets try to see if we can combine this into a single line, and avoid unnecessary use of creation of new variables